ucey-star/ID8
# personal-details hookup

## Description
This PR makes the personal details page functional. I have removed the form from the `app` directory and move it as a component to the `components folder`. This has solved issue https://github.com/ucey-star/ID8/issues/27 and now the personal-details page does not use the 'use client' as it did before. Additionally this PR closes #issue https://github.com/ucey-star/ID8/issues/35

How the changes work:
- We get the user from the page component and pass it as a prop to the personal details form. From this point we make a post request with the user information and redirect them to the home page.

Some changes:
- In the `personal-details` page I have removed the `name` and  `email` since we already get this from Google authentication. Consequently, I have changed the `name` to be `username` and the `age` to be `Date of Birth` since we want to dynamically update our users age on their birthday's. Currently, we have a database function that updates the user's ages everyday ensuring we also have accurate user ages.



## Type of Change


## Testing Instructions
- checkout into the branch 
- run `npm install` we have some new dependencies installed.
- Go the personal-details page input some information and if the post request is a a success you should be redirected to the home page. 

## Screenshots (if applicable)

If your changes include visual components, add screenshots showing the affected pages or elements.
